include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -0,0 +1,775 @@
+  /**
+   * Converting constructor from `layout_left_padded::mapping`.
+   *
+   * This overload participates in overload resolution only if `is_constructible_v<extents_type, OtherExtents>` is true.
+   * Either `padding_stride` or `OtherPaddingStride` must be `std::dynamic_extent`, or `padding_stride == OtherPaddingStride`.
+   */
+  MDSPAN_TEMPLATE_REQUIRES(
+    class _Mapping,
+    /* requires */ (
+      detail::__is_layout_left_padded_mapping<_Mapping>::value
+      && is_constructible_v<extents_type, typename _Mapping::extents_type>
+    )
+  )
+  MDSPAN_CONDITIONAL_EXPLICIT((extents_type::rank() > 1 && (padding_stride == dynamic_extent || detail::__padded_layout_padding_stride<typename _Mapping::layout_type>::value == dynamic_extent)))
+  constexpr
+  mapping(const _Mapping &__other_mapping)
+      : __inner_mapping(detail::__inner_extents_type<__extent_to_pad_idx, extents_type, __actual_padding_stride>::template __construct_other<padding_stride>(__other_mapping.extents(), __padding_stride_idx, __other_mapping)),
+        __unpadded_extent(detail::__unpadded_extent_type_impl<__extent_to_pad_idx, extents_type>::__construct(__other_mapping.extents()))
+  {
+    static_assert(padding_stride == dynamic_extent
+                  || detail::__padded_layout_padding_stride<typename _Mapping::layout_type>::value == dynamic_extent
+                  || padding_stride == detail::__padded_layout_padding_stride<typename _Mapping::layout_type>::value);
+  }
+
+  /**
+   * Converting constructor from `layout_right_padded::mapping`.
+   *
+   * This overload participates in overload resolution only if `extents_type::rank()` is 0 or 1 and `is_constructible_v<extents_type, OtherExtents>` is `true`.
+   */
+  MDSPAN_TEMPLATE_REQUIRES(
+    class _Mapping,
+    /* requires */ (
+      detail::__is_layout_right_padded_mapping<_Mapping>::value
+      && extents_type::rank() <= 1
+      && is_constructible_v<extents_type, typename _Mapping::extents_type>
+    )
+  )
+  MDSPAN_CONDITIONAL_EXPLICIT((!is_convertible_v<typename _Mapping::extents_type, extents_type>))
+  constexpr
+  mapping(const _Mapping &__other_mapping) noexcept
+      : __inner_mapping(__other_mapping.extents()),
+        __unpadded_extent(detail::__unpadded_extent_type_impl<__extent_to_pad_idx, extents_type>::__construct(__other_mapping.extents()))
+  {}
+
+  constexpr extents_type extents() const noexcept
+  {
+    if constexpr (extents_type::rank() == size_t(0))
+    {
+      return {};
+    } else {
+      return detail::__substitute_extents<__extent_to_pad_idx, __inner_extents_type, __unpadded_extent_type::static_extent(0)>::template __construct_with_type<extents_type>(__inner_mapping.extents(), __unpadded_extent);
+    }
+  }
+
+  constexpr std::array<index_type, extents_type::rank()>
+  strides() const noexcept
+  {
+    // Some compilers are over-zealous in their requirements checking on stride here
+    if constexpr ( extents_type::rank() == 0 )
+    {
+      return {};
+    } else {
```

Comment:
Would you consider this minor change for formatting consistency?

```suggestion
    if constexpr ( extents_type::rank() == 0 ) {
      return {};
    } else {
```